torch/library.py

```diff
@@ -954,6 +954,52 @@ def register(func):
         return register(func)
 
 
+def register_vmap(
```

Comment:
Btw, some good test cases are adding a register_vmap to all the custom ops in custom_op_db: https://github.com/pytorch/pytorch/blob/main/torch/testing/_internal/custom_op_db.py#L31

Reply:
I think some operators in op_db do not support vmap? For example, `numpy_cat(xs: Sequence[Tensor], dim: int)`. Running it on vmap we get error below because it has non-Tensor arguments. In this case do we just skip it? Maybe we should also add a check to check the schema when registering the vmap.

```
ValueError: vmap(f, in_dims=0, ...)(<inputs>): Got in_dim=0 for an input but the input is of type <class 'int'>. We cannot vmap over non-Tensor arguments, please use None as the respective in_dim
```

Reply:
You want to use in_dims=None at non-tensor indices, so: vmap(f, in_dims=(0, None))

Reply:
The testing strategy is to add a register_vmap to all of those custom ops, and then add them to this list so they automatically get tested (https://github.com/pytorch/pytorch/blob/f44739cf42e22a569bd1bdb0c113f8a069c17a41/test/functorch/test_vmap.py#L4166)

Reply:
I see, thanks!

Reply:
added now!